src/Symfony/Component/Workflow/MarkingStore/MethodMarkingStore.php

```diff
@@ -15,45 +15,42 @@
 use Symfony\Component\Workflow\Marking;
 
 /**
- * MethodMarkingStore stores the marking with a subject's method.
+ * MethodMarkingStore stores the marking with a subject's public method if exist,
+ * then to a property.
  *
  * This store deals with a "single state" or "multiple state" Marking.
  *
  * "single state" Marking means a subject can be in one and only one state at
- * the same time. Use it with state machine.
+ * the same time. Use it with state machine. It uses a string to store the marking
  *
  * "multiple state" Marking means a subject can be in many states at the same
- * time. Use it with workflow.
+ * time. Use it with workflow. It uses an array to store the marking
  *
  * @author Grégoire Pineau <lyrixx@lyrixx.info>
  */
 final class MethodMarkingStore implements MarkingStoreInterface
 {
-    private bool $singleState;
-    private string $property;
+    private readonly \SplObjectStorage $getters;
+    private readonly \SplObjectStorage $setters;
```

Comment:
we should use an array indexed by class instead, so that we don't keep subjects in memory
the closures you want should take the subject as argument to make this work

Reply:
I don't store the callable anymore, so the change is really simple!